### PR TITLE
doc: Asset Tracker v2: adding modules wrapper page

### PR DESCRIPTION
**Pull request by peknis**
To clean up the ToC for the Asset Tracker v2 docs, added a separate wrapper page for the modules.

### CODE REVIEW

File: applications/asset_tracker_v2/doc/firmware_architecture.rst [hunk 1]
```diff
@@ -24,18 +24,7 @@ It also shows the modules with thread and the modules without thread.
...
-* :ref:`asset_tracker_v2_data_module`
-* :ref:`asset_tracker_v2_cloud_module`
-* :ref:`asset_tracker_v2_sensor_module`
-* :ref:`asset_tracker_v2_location_module`
-* :ref:`asset_tracker_v2_ui_module`
-* :ref:`asset_tracker_v2_util_module`
-* :ref:`asset_tracker_v2_debug_module`
-* :ref:`asset_tracker_v2_modem_module`
-
+The application comprises of a set of internal modules.
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/doc/firmware_architecture.rst:27]: **question:** why is this changed from "modules" to "internal modules"?

Comment (coderbyheart) [reply applications/asset_tracker_v2/doc/firmware_architecture.rst:27]: @simensrostad can answer that, but I'd say do not change the meaning of the docs in this PR.

Comment (peknis) [reply applications/asset_tracker_v2/doc/firmware_architecture.rst]: Thanks for the nit, I like it. I was totally focused on the structural change and overlooked that. Fixed now.

File: applications/asset_tracker_v2/doc/modules.rst [hunk 2]
```diff
@@ -0,0 +1,26 @@
+.. _asset_tracker_v2_modules:
+
+Application modules
+###################
+
+The Asset Tracker v2 application uses a set of modules.
+See :ref:`asset_tracker_v2_internal_modules` for more information.
```
Comment (bjda) [inline applications/asset_tracker_v2/doc/modules.rst:7]: The tag for the architecture page should probably be renamed to `asset_tracker_v2_architecture`, especially since we now have another one called `asset_tracker_v2_modules`. I don't think it is linked elsewhere, and the current name is not precise.

The linking sentence could maybe be clarified a bit, like:
```suggestion
See :ref:`asset_tracker_v2_architecture` for more information on how the application is structured.
```

**Review by bjda — APPROVED** [review]
Looks good to me

File: applications/asset_tracker_v2/doc/modules.rst [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+   cloud_module
+   cloud_wrapper
+   debug_module
+   location_module
+   modem_module
+   sensor_module
+   ui_module
+   util_module
+   modules_common
+   unit_test
```
[the remark this answers is not in the document]
Comment (peknis) [reply applications/asset_tracker_v2/doc/modules.rst:26]: Good point. Let me see what to do about it.
